### ISSUE TITLE
Error running WaveTransmittedTurbulence.jl with julia 1.9.3

### ISSUE THREAD

**Issue by Swarnali-D** [glwagner/WaveTransmittedTurbulence#1]
I am trying to reproduce the figures of WaveTransmittedTurbulence.jl in julia version 1.9.3
On running `julia --project -e 'using Pkg; Pkg.instantiate()`', I am met with an error :

`┌ Warning: The active manifest file is an older format with no julia version entry. Dependencies may have been resolved with a different julia version. └ @ ~/WaveTransmittedTurbulence/Manifest.toml:0 ERROR: Could not locate the source code for the DelimitedFiles package. Are you trying to use a manifest generated by a different version of Julia? `

How do I overcome this? Or is there any other way I can run the simulations?

**Comment by glwagner** [comment]
The first thing to try is to delete the Manifest and instantiate again.

**Comment by glwagner** [comment]
Unfortunately there are no compat entries and this will surely fail.

**Comment by glwagner** [comment]
You can try this PR: 

https://github.com/glwagner/WaveTransmittedTurbulence.jl/pull/2

I added a compat entry and deleted the Manifest.

Can I ask your intent / goals? That will help us understand what to do about it. If you need the precise scientific results for some reason, we have to reproduce the setup. If you just want the general idea / setup, it's better to start with a fresh script in fact, because this is not very challenge to reproduce. Also it depends which simulation you want to reproduce. One of them is maintained as an example in Oceananigans.

**Comment by Swarnali-D** [comment]
I tried to run the example code in Oceananigans and tried to see if the figures matches with the resting cases of figure 7 of your article. It does not. So I want to run this repository and see if I'm able to reproduce the results. 
I want to reproduce the steady wave case, it would be helpful if you can tell me what's the difference in the Oceananigans code and the repository code that was not able to match the plots.

**Comment by glwagner** [comment]
The main difference is the resolution. Can you tell me if you used the example at its given resolution, or whether you increased the resolution to match the paper?

The numerics may also differ --- check the advection scheme and turbulence closure. I'll take a look when I can.

**Comment by Swarnali-D** [comment]
I gave a resolution of 256^3. 
advection= WENO()
closure = AnisotropicMinimumDissipation()

In the Oceananigans example, there is no sponge layer, whereas in your article there is a sponge layer as far as I have understood. Can this be one of the reasons?

**Comment by glwagner** [comment]
Hmm. The sponge layer could matter but I think the effect should be pretty minor. More important perhaps is that the paper uses `advection = Centered(order=2)`. You can also try running with `closure=nothing` and `advection=WENO(order=9)`. `advection=WENO()` defaults to 5th order which is quite diffusive.

Can you be specific or provide a plot showing how your simulation differs from the paper?

**Comment by Swarnali-D** [comment]
Thank you for providing me with the used advection scheme.  I can try with `closure=nothing` but the paper uses `closure=AnisotropicMinimumDissipation`, I just want to try the exact model parameters. 

![w_variance](https://github.com/user-attachments/assets/c5199415-4bba-48e7-a34c-101d48a9007c)
![speed](https://github.com/user-attachments/assets/b6a3156b-176f-410a-b348-dd28ae37cb35)
![gradB](https://github.com/user-attachments/assets/15a7a579-5e31-4666-958a-b8debee3d52d)
![buoyancy](https://github.com/user-attachments/assets/6bf6123c-d7a0-4f8c-85ab-bb9f270dfb45)

**Comment by glwagner** [comment]
Ah! Check the initial buoyancy profile. They are different.

**Comment by glwagner** [comment]
Ok there are two things. First, in the plot you shared it seems there is a discrepancy in the initial buoyancy gradient. Second, the example has an initial mixed layer whereas the epaper case does not. Can you try fixing these two things and see what happens?

**Comment by Swarnali-D** [comment]
The buoyancy parameters in the example (which I used) are: 
```
Qᵇ = 2.307e-8 # m² s⁻³, surface buoyancy flux
N² = 1.936e-5 # s⁻², initial and bottom buoyancy gradient
```

And the paper has these parameters: 
![Screenshot from 2024-09-14 11-15-24](https://github.com/user-attachments/assets/296ce9c9-1e08-483a-a313-b6160f288b38)
Am I right? 
I just noticed that the paper doesn't have an initial mixed layer, I will fix that. But why did you include that in the example?
Also, you did a spin up simulation which created a mixed layer of z=-8m, didn't you use this as initial condition for the resting case?

**Comment by glwagner** [comment]
An initial mixed layer helps speed up the simulation. It's not a bad idea to use that in general. The paper and example have different goals.

**Comment by glwagner** [comment]
The spin up simulation is only used for the first part of Wagner et al. 2021

**Comment by glwagner** [comment]
As for the buoyancy profile, the point is just that the buoyancy profile below the mixed layer is different between the two simulations. Since this part of the domain shouldn't change it has to mean the initial condition is different. I'm not sure how you make that plot though.

**Comment by glwagner** [comment]
@Swarnali-D were you able to reproduce the results to your satisfaction?

**Comment by Swarnali-D** [comment]
Not really, I was not able to run this repository on my Oceananigans version. I tried an older version (v.0.54.2) which has IncompressibleModel (same as the one your article uses) and tried to reproduce [McWilliams et al.(1997)](https://www.cambridge.org/core/journals/journal-of-fluid-mechanics/article/langmuir-turbulence-in-the-ocean/638FD0E368140E5972144348DB930A38) article. It does to some extent. But I am not sure if the old version validation is still true with the stable version since you haven't mentioned in the documentation that the stable version validates McWilliams et al.(1997) .  
![McW_stats](https://github.com/user-attachments/assets/f8c6960e-16f9-4ec0-a111-c7d4517a5c6b)

**Comment by Swarnali-D** [comment]
Also the Eulerian-mean u velocity profiles in your article (Figure A1. (a)) and McWilliams et al. (1997) (Figure 2. S/0.3 case) don't seem to match. Am I missing something here? Are you plotting instantaneous profiles or time-averaging?
![Screenshot 2024-11-11 at 16-53-11 Langmuir turbulence in the ocean - langmuir-turbulence-in-the-ocean pdf](https://github.com/user-attachments/assets/f01b5f3b-4308-4fa7-bd0f-bb00552f6c5b)
![Screenshot from 2024-11-11 16-51-59](https://github.com/user-attachments/assets/10820fab-343a-4920-a54f-4c77c3266889)

**Comment by glwagner** [comment]
I guess they are both mixed but there is a small offset in McWilliams et al 1997. Is that what you are referring to?

The turbulence closures are different between the two papers. In particular, McWilliams et al 1997 use an SGS model that fluxes momentum down the Eulerian-mean gradient. Oceananigans fluxes momentum down the Lagrangian-mean gradient. Wagner et al 2021 (this repo) shows how the McWilliams et al model is not well-posed in general; ie it can act as a source of KE.

It could be that the difference in SGS model causes this discrepancy (or perhaps there is another bug that we have not found). In terms of the SGS discrepancy, we could potentially try to correct it --- though I would argue that the model in McWilliams et al 1997 is incorrect.

**Comment by Swarnali-D** [comment]
Thanks for the explanation, that cleared it up! I appreciate your help.
